angrmanagement/plugins/log_human_activities/log_human_activities.py

```diff
@@ -0,0 +1,136 @@
+            project_md5=self.project_md5,
+            old_name=old_name,
+            new_name=new_name,
+            created_by=TODO,
+        )
+        self.session.add(function_rename)
+        self.session.commit()
+        l.info("Add function rename sesssion to slacrs, project name %s, old_name %s, new_name %s", self.project_name, old_name, new_name)
+        # result = self.session.query(HumanActivityFunctionRename).filter(HumanActivityFunctionRename.project == self.project_name).first()
+        # l.info("Query result: old_name %s, new_name %s", result.old_name, result.new_name)
+
+    def handle_click_block(self, qblock, event):
+        block_click = HumanActivityClickBlock(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            addr=qblock.addr,
+            created_by=TODO,
+        )
+        self.session.add(block_click)
+        self.session.commit()
+        l.info("Block %x is clicked", qblock.addr)
+        return False
+
+    def handle_click_insn(self, qinsn, event):
+        insn_click = HumanActivityClickInsn(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            addr=qinsn.addr,
+            created_by=TODO,
+        )
+        self._submit_to_slacrs(insn_click)
+        l.info("Instruction %x is clicked", qinsn.addr)
+        return False
+
+    def handle_raise_view(self, view):
+        # e.g., "<class 'angrmanagement.ui.views.disassembly_view.DisassemblyView'>"
+        view_name = str(view.__class__).split('.')[-1][:-2]
+        raise_view = HumanActivityRaiseView(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            view=view_name,
+            created_by=TODO,
+        )
+        self._submit_to_slacrs(raise_view)
+        l.info("View %s is raised", view_name)
+
+    def handle_comment_changed(self, addr: int, cmt: str, new: bool, decomp: bool):
+        """
+        Log a user's activity of changing comment
+        @param new: T if a new comment. We don't log it in slacrs.
+        @param comp: T if comment is in decompiler view
+        """
+        comment_change = HumanActivityCommentChanged(
+            project=self.project_name,
+            project_md5=self.project_md5,
+            addr=addr,
+            cmt=cmt,
+            decomp=decomp,
+            created_by=TODO,
+        )
+        self.session.add(comment_change)
+        self.session.commit()
+        l.info("Comment is added at %x", addr)
+        return False
+
+    def handle_project_initialization(self):
+        """
+        Set project name
+        """
+        if self.workspace.instance.img_name is not None:
+            self.project_name = self.workspace.instance.img_name
+        else:
+            filename = self.workspace.instance.project.filename
+            self.project_name = filename
+            with open(filename, 'rb') as f:
+                self.project_md5 = hashlib.md5(f.read()).hexdigest()
```

Comment:
Are you sure this works when angr management loads an `adb` instead of a binary?

You probably want to use `self.workspace.instance.project.loader.main_object.sha256` or `self.workspace.instance.project.loader.main_object.md5` (I'm not sure if `.md5` is a thing),